### DISCUSSION TITLE
Saturation

### DISCUSSION THREAD

**Discussion by marinsiebert** [idaholab/moose#23335] [Q&A General] [answered]
Hello everyone, 

I am encountering an issue, which for the love of me I am not able to fix properly. 
I read on multiple occasions that the saturation calculation within the porous flow module considers a model to be fully saturated (S = 1.0) when the porepressure (Pp) is above 0, therefore S should be <1 when  Pp <= 0, however I cannot reproduce this assumption.
I tried it with the THM, Fully Saturated and Unsaturated Action as well as with my own Kernel setup. 
The model is supposed to be a 2D soil environment which I want to investigate in regards of thermal fields in a dry and fully water saturated, respectively, as well as at differents level of residual saturation and water tables. I assume that my comprehension of the problem is wrong at some point but I cannot figure out where, so hopefully someone might be able to help me out.
Below an example for the THM action.  Note that here the porepressures are far below zero (which of course does not make sense), however the saturation is still remaining at 1. What am I missing? 

[dummy.zip](https://github.com/idaholab/moose/files/10580424/dummy.zip)
```
[Mesh]
  [./mesh]
    type = FileMeshGenerator
    file = mesh.off
    allow_renumbering = false
  [../]
  [./outer]
    type = SideSetsFromNormalsGenerator
    input = mesh
    new_boundary = 'top left bottom right'
    normals = '0 1 0 -1 0 0 0 -1 0 1 0 0'
  [../]
  [./cable]
    type = SubdomainBoundingBoxGenerator
    input = outer
    block_id = '1'
    bottom_left = '-0.06 -1.55 0'
    top_right = '0.045 -1.44 0'


  [../]
  [./blocks]
    type = RenameBlockGenerator
    input = cable
    old_block = '0 1'
    new_block = 'soil cable'
  [../]
[]

[GlobalParams]
  biot_coefficient = 1
  PorousFlowDictator = dictator
  displacements = 'disp_x disp_y'
[]

[Functions]
  [./pressure_gradient]
    type = ParsedFunction
    vars = 'a b'
    vals = '1000 -9.8065'
    value = 'if(y>-1.5,0, a*b*y)'
  [../]
  [./temperature_gradient]
    type = ParsedFunction
    vars = 'a c'
    vals = '-0.03 293.15'
    value = a*y+c
  [../]
[]

[Variables]
  [./temperature]
    [./InitialCondition]
      type = FunctionIC
      variable = temperature
      function = temperature_gradient
    [../]
    scaling = 1e-5
  [../]
  [./porepressure]
    [./InitialCondition]
      type = FunctionIC
      variable = porepressure
      function = pressure_gradient
    [../]
  [../]
  [./disp_x]
    scaling = 1e-10
  [../]
  [./disp_y]
    scaling = 1e-10
  [../]
[]

[PorousFlowBasicTHM]
  porepressure = porepressure
  temperature = temperature
  coupling_type = ThermoHydroMechanical
  gravity = '0 0 0'
  fp = simple_fluid
  eigenstrain_names = thermal_contribution
  use_displaced_mesh = false
  displacements = 'disp_x disp_y'
[]

[BCs]

  [./pinned_topx]
   type = DirichletBC
   variable = disp_x
   value = 0
   boundary = 'top bottom'
  [../]
  [./roller_tmax]
    type = DirichletBC
    variable = disp_x
    value = 0
    boundary = 'left right'
  [../]
  [./pinned_topy]
   type = DirichletBC
   variable = disp_y
   value = 0
   boundary = 'top'
  [../]
 []
[]

[FluidProperties]
  [simple_fluid]
    type = SimpleFluidProperties
    bulk_modulus = 2e9
    viscosity = 0.001
    density0 = 1000
    thermal_expansion = 0.000214
    cp = 4194
    cv = 4186
    porepressure_coefficient = 1
  [../]
[]



[Materials]


#_______________________________________________

#General

#_______________________________________________

[./porosity_soil]
  type = PorousFlowPorosity
  porosity_zero = 0.1
[../]
#_______________________________________________

#Thermo

#_______________________________________________

[./thermal_conductivity_soil]
  type = PorousFlowThermalConductivityIdeal
  dry_thermal_conductivity = '0.4 0 0  0 0.4 0  0 0 0.4'
  wet_thermal_conductivity = '1 0 0 0 1 0 0 0 1'
[../]

[./internal_energy_soil]
  type = PorousFlowMatrixInternalEnergy
  density = 1600
  specific_heat_capacity = 835
[../]

[./thermal_expansion_soil]
  type = PorousFlowConstantThermalExpansionCoefficient
  drained_coefficient = 0.003
  fluid_coefficient = 0.0002
[../]
#_______________________________________________

#Hydro

#_______________________________________________

[./permeability_soil]
  type = PorousFlowPermeabilityConst
  permeability = '1e-14 0 0   0 1e-14 0   0 0 1e-14'
[../]

#_______________________________________________

#Mechanical
#_______________________________________________

[./biotconstant_uni]
  type = PorousFlowConstantBiotModulus
  solid_bulk_compliance = 2.85e-11
  fluid_bulk_modulus = 2e9
[../]

[./elasticity_soil]
  type = ComputeIsotropicElasticityTensor
  shear_modulus = 44e9
  bulk_modulus = 38e9
[../]

[./thermal_contribution_soil]
  type = ComputeThermalExpansionEigenstrain
  temperature = temperature
  thermal_expansion_coeff = 0.001
  eigenstrain_name = thermal_contribution_soil
  stress_free_temperature = 293.15
[../]

[./stress_uni]
  type = ComputeLinearElasticStress
[../]

[./strain_soil]
  type = ComputeSmallStrain #
  eigenstrain_names = 'thermal_contribution_soil'
[../]

[Preconditioning]
  active = basic
  [./basic]
    type = SMP
    full = true
    petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
    petsc_options_value = ' lu       mumps'
  [../]
  [./extended]
   type = SMP
   full = true
   petsc_options = '-ksp_diagonal_scale -ksp_diagonal_scale_fix'
   petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
   petsc_options_value = ' asm      lu           NONZERO                   2'
  [../]
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  end_time = 1
  dt = 1
  nl_abs_tol = 1e-10
  nl_rel_tol = 1e-10
[]

[Outputs]
  print_linear_residuals = true
  perf_graph = true
  [./exodus]
  type = Exodus
  file_base = output
  [../]
  [./csv]
  type = CSV
  file_base = output
  [../]

[]
```

**Comment by cpgr** [comment]
The `PorousFlowBasicTHM` assumes that the model is fully saturated - see https://mooseframework.inl.gov/modules/porous_flow/additional_objects.html

But you say that you tried using `PorousFlowUnsaturated`? That should have worked?

**Comment by marinsiebert** [comment, marked as answer]
Well this is awkward, I must have overread this. However, the same happens in PorousFlowUnsaturated. I tried it without BCs as well as with BCs at all boundaries with zero pressure using a Dirichlet and a Neumann, respectively. The latter does create some gradient artifacts, the values remain the same though. I am able to create a reduction in the saturation by emptying the model with sinks through the boundaries, but this does not dry out the whole model. 

[Unsaturated.zip](https://github.com/idaholab/moose/files/10607911/Unsaturated.zip)

[dry_steady.zip](https://github.com/idaholab/moose/files/10607913/dry_steady.zip)

[Variables]
  [./temperature]
    [./InitialCondition]
      type = FunctionIC
      variable = temperature
      function = temperature_gradient
    [../]
    scaling = 1e-8
  [../]
  [./porepressure]
    [./InitialCondition]
      type = ConstantIC  #FunctionIC
      value = 0
      variable = porepressure
    [../]
  [../]
[]

**Reply by marinsiebert** [reply]
Apparently my misinterpretation happened in the [PorousFlowCapillaryPressureVG](https://mooseframework.inl.gov/moose/source/userobjects/PorousFlowCapillaryPressureVG.html), which clearly states, that the effective saturation sits at 1.0 for Pp >= 0, therefore Pp has to be negative  (not = 0) with Pp = - Pc (capillary pressure). 
In order for the saturation to decrease one then has to choose the van genuchten parameters m and alpha as well as Pp accordingly, e.g. Pp = 1e-6 m = 0.3 and alpha = 1e-4 yields a saturation of roughly 0.14, an Excel-Sheet is helpful to calulcate the achievable saturations. 

Assuming this is now correct I hope this helps others who encounter the same problem.
